cpp/mcap/include/mcap/mcap.hpp

```diff
@@ -696,6 +701,62 @@ struct TypedRecordReader {
   bool parsingChunk_;
 };
 
+struct LinearMessageView {
+  struct Iterator {
+    using iterator_category = std::forward_iterator_tag;
+    using difference_type = int64_t;
+    using value_type = Message;
+    using pointer = const Message*;
+    using reference = const Message&;
+
+    reference operator*() const;
+    pointer operator->();
```

Comment:
```suggestion
    pointer operator->() const;
```
any reason this doesn't match `operator*`?